packages/legacy/core/App/hooks/initialize-agent.ts

```
@@ -101,6 +125,12 @@ const useInitializeAgent = () => {
       return
     }
 
+    const existingAgent = await restartExistingAgent()
+    if (existingAgent) {
+      setAgent(existingAgent)
+      return existingAgent
+    }
+
     const newAgent = await createNewAgent()
     if (!newAgent) {
       return
```

no agent to restart and it failed to create a new agent, this should probably throw an exception or at least log